e2e/nomostest/config-sync.go

```diff
@@ -511,15 +513,60 @@ func validateMultiRepoDeployments(nt *NT) error {
 		return errs
 	}
 
-	// Validate the webhook config seperately, since it's not a Deployment.
-	took, err := Retry(nt.DefaultWaitTimeout, func() error {
-		return nt.Validate("admission-webhook.configsync.gke.io", "", &admissionv1.ValidatingWebhookConfiguration{})
-	})
-	if err != nil {
-		return err
+	// Validate the webhook config separately, since it's not a Deployment.
+	if !nt.WebhookDisabled {
+		took, err := Retry(nt.DefaultWaitTimeout, func() error {
+			return nt.Validate("admission-webhook.configsync.gke.io", "", &admissionv1.ValidatingWebhookConfiguration{})
+		})
+		if err != nil {
+			return err
+		}
+		nt.T.Logf("took %v to wait for %s %s", took, "ValidatingWebhookConfiguration", "admission-webhook.configsync.gke.io")
 	}
-	nt.T.Logf("took %v to wait for %s %s", took, "ValidatingWebhookConfiguration", "admission-webhook.configsync.gke.io")
 
+	return validateMultiRepoPods(nt)
+}
+
+// validateMultiRepoPods validates if all Config Sync Pods are healthy. It doesn't retry
+// because it is supposed to be invoked after validateMultiRepoDeployments succeeds.
+func validateMultiRepoPods(nt *NT) error {
```

Comment:
This looks like it replicates what's happening in kstatus StatusReader, which has exceptions for checking Deployment > ReplicaSet > Pod and lifting out extra status details.

Reply:
Do you mean creating a custom StatusReader for the Config Sync Pods? I feel like the validation is specific to the test, e.g., we only check the Pods without DeletionTimestamp that have no container terminated because of OOMKilled.